CMakeLists.txt

```diff
@@ -45,18 +45,49 @@ if(POLICY CMP0135)
   cmake_policy(SET CMP0135 NEW)
 endif()
 
-# Check if any relevant env vars were set from the build env scripts
-if(DEFINED ENV{X_VCPKG_APPLOCAL_DEPS_INSTALL} AND NOT DEFINED X_VCPKG_APPLOCAL_DEPS_INSTALL)
-  set(X_VCPKG_APPLOCAL_DEPS_INSTALL "$ENV{X_VCPKG_APPLOCAL_DEPS_INSTALL}" CACHE BOOL "")
+# Note: ENV{VCPKG_ROOT} is the default location for the vcpkg cli tool which is
+# later adjusted by CMAKE_TOOLCHAIN_FILE. We use it here as a workaround to find the
+# overlay folders in manifest mode https://github.com/microsoft/vcpkg/issues/12289.
+if(DEFINED ENV{VCPKG_ROOT} AND NOT DEFINED VCPKG_ROOT)
+  if(EXISTS "$ENV{VCPKG_ROOT}/overlay/ports" OR NOT EXISTS "$ENV{VCPKG_ROOT}/ports")
+    # We assume ENV{VCPKG_ROOT} points to our vcpkg environment
+    # and we configure the CMAKE_TOOLCHAIN_FILE and overlays accordingly
+    message(STATUS "Setting VCKG_ROOT from environment: $ENV{VCPKG_ROOT}")
+    set(VCPKG_ROOT "$ENV{VCPKG_ROOT}")
+  else()
+    # The official GitHub Actions runner images sometimes set these
+    # variables, but we don't want to use vcpkg from the runner. If we use
+    # vcpkg, we use our own prebuilt environment, not the one from the image.
+    message(STATUS "Ignoring env VCKG_ROOT (missing $ENV{VCPKG_ROOT}/overlay/ports)")
+  endif()
 endif()
 
-if(DEFINED ENV{VCPKG_DEFAULT_TRIPLET} AND NOT DEFINED VCPKG_TARGET_TRIPLET)
-  set(VCPKG_TARGET_TRIPLET "$ENV{VCPKG_DEFAULT_TRIPLET}" CACHE STRING "")
+if(DEFINED VCPKG_ROOT)
+  if(NOT DEFINED VCPKG_OVERLAY_PORTS)
+    # required for manifest mode
+    set(VCPKG_OVERLAY_PORTS "${VCPKG_ROOT}/overlay/ports")
+    if(APPLE)
+        list(APPEND VCPKG_OVERLAY_PORTS "${VCPKG_ROOT}/overlay/osx")
+    elseif(WIN32)
+        list(APPEND VCPKG_OVERLAY_PORTS "${VCPKG_ROOT}/overlay/windows")
+    endif()
+  endif()
+
+  if(NOT DEFINED VCPKG_OVERLAY_TRIPLETS)
+    # required for manifest mode
+    set(VCPKG_OVERLAY_TRIPLETS "${VCPKG_ROOT}/overlay/triplets")
+  endif()
```

Comment:
This is something mixxx-specific, isn't it? If so, we should probably only set it when taking the VCPKG_ROOT from the `MIXXX_VCPKG_ROOT` environment variable. Otherwise it may interfere with custom VCPKG setups.

Reply:
Your comment is on the state of the PR before my last commit. I don't know why GitHub does not mark it as outdated. 
Your suggestion is done in the final commit. We are now entirely VCPKG_ROOT independent.